### ISSUE TITLE
Debugging Clarity - 'ModelDeclarativeMetaclass' has no attribute 'Meta' when passing type rather than instance

### ISSUE THREAD

**Issue by kylesquared** [django-tastypie/django-tastypie#1353]
I ran into a bit of a confusing error message today. I added a few model resources to API, and suddenly everything broke for a few of the endpoints. 

I got the following error message
```
AttributeError at /crm/work-orders/15/
type object 'ModelDeclarativeMetaclass' has no attribute 'Meta'
```

It turns out that this was a really simple issue. I had forgot to instantiate model resources I was registering.

 Changing from:
```python
version_1_api = Api(api_name='v1')
version_1_api.register(WorkOrderResource)
```

to:
```python
version_1_api = Api(api_name='v1')
version_1_api.register(WorkOrderResource())
``` 

fixed the issue for me. However, this solution was not obvious, and there was no real clear indication that this was the issue, short of reading the source.

I know this goes against what the documentation states, but it was a simple mistake in a large file, which took a bit of time to figure out.

Can an exception be added, when registering an endpoint, that checks to see if it is an instance or a type, and clearly notify the user of what the issue is?

**Comment by kylesquared** [comment]
Here is the line in question:

```
/lib/python2.7/site-packages/tastypie/api.py in register
            if resource_name in self._canonicals:
                warnings.warn("A new resource '%r' is replacing the existing canonical URL for '%s'." % (resource, resource_name), Warning, stacklevel=2)
            self._canonicals[resource_name] = resource
            # TODO: This is messy, but makes URI resolution on FK/M2M fields
            #       work consistently.
            resource._meta.api_name = self.api_name
                        resource.__class__.Meta.api_name = self.api_name 
```

**Comment by SeanHayes** [comment]
We should probably raise a ValueError explaining an instance needs to be passed in.

Would you be interested in submitting a PR with a test case? If not I can get to it later.

**Comment by SeanHayes** [comment]
Fixed by #1361
